### PR TITLE
Added various functions basic math classes. Also enabled math checks …

### PR DESCRIPTION
**Pull request by ghost**
…only for debug builds.

Added set_scale, set_rotation_euler, set_rotation_axis_angle. Addresses #2565 directly.
Added an euler angle constructor for Basis in GDScript and also exposed is_normalized for vectors and quaternions.
Various other changes mostly cosmetic in nature.

### CODE REVIEW

**Comment by tagcup** [comment]
Some references:
In #8214, @reduz suggested me to enable the mathematical sanity check (is_normalized check for Vector2 and Vector3) only in debug builds. This PR consistently aligns the rest of core/math by hiding mathematical checks behind MATH_CHECKS, only defined for debug builds.

Also addresses @n-pigeon request about `is_normalized` in #8214.

File: core/math/math_2d.cpp [hunk 1]
```diff
@@ -62,7 +62,8 @@ Vector2 Vector2::normalized() const {
 }
 
 bool Vector2::is_normalized() const {
-	return Math::isequal_approx(length(), (real_t)1.0);
+	// use length_squared() instead of length() to avoid sqrt(), makes it more stringent.
+	return Math::is_equal_approx(length_squared(), 1.0);
```
Comment (bojidar-bg) [inline core/math/math_2d.cpp]: You can actually drop this comment

File: core/variant_call.cpp [hunk 2]
```diff
@@ -1748,6 +1765,7 @@ void register_variant_methods() {
 
 	_VariantCall::add_constructor(_VariantCall::Basis_init1, Variant::BASIS, "x_axis", Variant::VECTOR3, "y_axis", Variant::VECTOR3, "z_axis", Variant::VECTOR3);
 	_VariantCall::add_constructor(_VariantCall::Basis_init2, Variant::BASIS, "axis", Variant::VECTOR3, "phi", Variant::REAL);
+	_VariantCall::add_constructor(_VariantCall::Basis_init3, Variant::BASIS, "euler", Variant::VECTOR3);
```
Comment (bojidar-bg) [inline core/variant_call.cpp]: Not sure about that... should it be euler or translation?

**Review by bojidar-bg — COMMENTED** [review]
Mostly LGTM.

**Comment by bojidar-bg** [comment]
Also, I assume you ran -doctool prior/after you changed the docs?

File: core/variant_call.cpp (hunk 2 above)
Comment (tagcup) [inline core/variant_call.cpp]: This is for Basis, not Transform. There is not translation.

File: core/math/math_2d.cpp (hunk 1 above)
Comment (tagcup) [inline core/math/math_2d.cpp]: I'd rather keep it because it's a subtlety: for small deviations from 1.0, it effectively makes the epsilon twice as small.

**Comment by tagcup** [comment]
Nope, I've never done that before, at all. Isn't that something the project members do at certain intervals, rather than whenever docs changed?

**Comment by tagcup** [comment]
I just tried running doctool, and it's giving me some changes related to other stuff recently merged such as DLScript, AStar, etc. Do you prefer if I committed those nevertheless?

**Comment by bojidar-bg** [comment]
@tagcup Ah, sorry then. If you prefer you can do it in a seperate commit (to reduce noise to some extent), or you can just leave it as-is, and I (or someone else) would sync it later on.

**Comment by tagcup** [comment]
Cool, although, let me quickly modify a comment about polar decomposition; I think it really deserves some explanation because it's a bit opaque for potential future contributors.

**Comment by tagcup** [comment]
Done, just expanded a few code comments.

**Comment by tagcup** [comment]
BTW, for future, I think it'd be nice if we could expose `Quat::get_axis_angle()`. Is there a way in GDScript to return multiple values, like passing reference (vector and real) arguments?

**Comment by tagcup** [comment]
@akien-mga To clarify the title, this PR doesn't add any new math checks. It rather disables those checks for non-debug builds.

**Review by akien-mga — APPROVED** [review]

**Comment by akien-mga** [comment]
Please handle your TODO about `get_rotation` needing to be renamed soon, so that we don't forget :)

**Comment by tagcup** [comment]
Yup, I just need @reduz's confirmation on that! I'll ask him over IRC

**Comment by tagcup** [comment]
@akien-mga I actually have a TODO list I keep maintaining [here](https://github.com/godotengine/godot/pull/6865#issuecomment-254934040), just added there to be safe

**Pull request merged**